src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -37,12 +39,14 @@ public partial class BlockProcessor(
     IBlockProcessor.IBlockTransactionsExecutor? blockTransactionsExecutor,
     IWorldState? stateProvider,
     IReceiptStorage? receiptStorage,
-    IBlockhashStore? blockHashStore,
+    ITransactionProcessor transactionProcessor,
     IBeaconBlockRootHandler? beaconBlockRootHandler,
+    IBlockhashStore? blockHashStore,
     ILogManager? logManager,
     IWithdrawalProcessor? withdrawalProcessor = null,
     IReceiptsRootCalculator? receiptsRootCalculator = null,
-    IBlockCachePreWarmer? preWarmer = null)
+    IBlockCachePreWarmer? preWarmer = null,
```

Comment:
I think `preWarmer` should be last as it is not domain level dependency but a very technical one.

Reply:
@rjnrohit